tests/test_models.py

```diff
@@ -269,3 +268,27 @@ def __init__(self, *args, **kwargs):
     model = FailedModel()
     # "Asserting" no error here:
     model.__del__()
+
+
+def test_on_init_hook():
+    class OnInitModel(DataModel):
+        def on_init(self, init):
+            super().on_init(init)
+
+            self.meta.foo = "bar"
+
+    model = OnInitModel()
+    assert model.meta.foo == "bar"
+
+
+def test_on_save_hook(tmp_path):
+    class OnSaveModel(DataModel):
+        def on_save(self, init):
+            super().on_save(init)
+
+            self.meta.foo = "bar"
+
+    model = OnSaveModel()
+    assert model.meta.get("foo") is None
```

Comment:
Why use `get()` here instead of 

```python
assert model.meta.foo is None
```
?

Reply:
`foo` isn't in the schema, so retrieving its value before it has been set will cause an error